polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -512,6 +523,21 @@ where
 	Some(processed_code_hashes)
 }
 
+async fn maybe_update_active_leaf(
+	mut validation_backend: impl ValidationBackend,
```

Comment:
In a future PR we can do even better. We keep track of last finalized block and then we can drop approval execution for the inclusion relay chain blocks that are already finalized.

Reply:
I doubt that improves anything.